bad_code.java

```diff
@@ -0,0 +1,54 @@
+public class BadCodeExample {
+    
+    public static void main(String[] args) {
+        ChildClass child = new ChildClass();
+        child.doSomething();
+        child.doSomethingElse();
+    }
+}
+
+class ParentClass {
+    public int someNumber = 10;
+    public String someString = "Hello";
+
+    public void doSomething() {
+        System.out.println("Doing something in ParentClass");
+        someNumber++;
+        someString += " Worlds";
+        System.out.println(someString);
+    }
+
+    public void doSomethingElse() {
```

Comment:
Refactor the `doSomethingElse` method to a shared method in `ParentClass` that accepts parameters or uses other mechanisms to handle the differences between the parent and child class implementations. This will reduce code duplication and improve maintainability. [important]